src/lib/scan/scan_content.js

```diff
@@ -0,0 +1,309 @@
+import {log} from '/lib/util/log.js';
+import {isMajorChange} from './fuzzy.js';
+import {Page} from '../page/page.js';
+
+/**
+ * Enumeration indicating the similarity of two HTML strings.
+ *
+ * @readonly
+ * @enum {string}
+ */
+export const changeEnum = {
+  NEW_CONTENT: 'new_content',
+  NO_CHANGE: 'no_change',
+  MAJOR_CHANGE: 'major_change',
+  MINOR_CHANGE: 'minor_change',
+};
+
+export const __ = {
+  log: (...args) => log(...args),
+  isMajorChange: (...args) => isMajorChange(...args),
+  changeEnum: changeEnum,
+  stripHtml: stripHtml,
+  getChanges: getChanges,
+  getIteratorFunction: getIteratorFunction,
+};
+
+/**
+ * Content data with previously prepared and chopped HTML.
+ *
+ * @property {string} html - Page HTML.
+ * @property {?Array} parts - HTML split.
+ */
+export class ContentData {
+  /**
+   * @param {string} html - Page HTML.
+   * @param {?Array} parts - HTML split.
+   */
+  constructor(html, parts = null) {
+    this.html = html || '';
+    this.parts = parts || null;
+  }
+
+}
+
+/**
+ * Detects changes based on two HTML data and page.
+ *
+ * @param {ContentData} prevData - Data for previous HTML.
+ * @param {ContentData} scannedData - Data for scanned HTML.
+ * @param {Page} page - Page.
+ *
+ * @returns {string|changeEnum} ChangeEnum string indicating how similar the
+ * two HTML strings are.
+ */
+export function getChanges(prevData, scannedData, page) {
+  if (prevData.html == null || prevData.html === '') {
+    return changeEnum.NEW_CONTENT;
+  }
+
+  if (page.requireExactMatchCount || page.requireExactMatchCount == null) {
+    const countChange = getCountChange(prevData, scannedData);
+    if (countChange === changeEnum.MAJOR_CHANGE) {
+      return countChange;
+    }
+  }
+
+  if (prevData.html === scannedData.html) {
+    return changeEnum.NO_CHANGE;
+  }
+
+  const contentModeEnum = Page.contentModeEnum;
+  const contentMode = page.contentMode || contentModeEnum.TEXT;
+  if (contentMode === contentModeEnum.IGNORE) {
+    return changeEnum.NO_CHANGE;
+  }
+
+  const prevParts = prevData.parts || [prevData.html];
+  const scannedParts = scannedData.parts || [scannedData.html];
+
+  const ignoreTags = contentMode !== contentModeEnum.HTML;
+
+  const htmlChange = getHTMLChange(page, prevParts, scannedParts, ignoreTags);
+
+  // If no change was detected in parts, just return minor change because we
+  // already know there was a change somewhere in the html
+  if (htmlChange === changeEnum.NO_CHANGE) {
```

Comment:
This is the check you wanted if any change is detected outside of the selected html elements than report it as minor change. And it needs to strip the tags so that it only compares text.

Reply:
Of course - thanks!